File: php/010-class.php
<?php
interface ExampleInterface
{
    public function foo(): string;
}

class Base
{
    protected $bar = 'bar';
}

class Example extends Base implements ExampleInterface
{
    public function __construct()
    {
        echo 'initialize';
    }

    public function foo(): string
    {
        return $this->bar;
    }
}

$example = new Example();

echo $example->foo();

File: swift/010-class.swift
protocol ExampleProtocol {
    func foo() -> String
}

class Base {
    internal let bar = "bar"
}

class Example: Base, ExampleProtocol {
    override init() {
        print("initialize")
    }

    func foo() -> String {
        return self.bar
    }
}

let example = Example() // no need 'new'
print(example.foo())
